engine/src/main/java/io/zeebe/engine/processing/message/ProcessMessageSubscriptionCorrelateProcessor.java

```diff
@@ -104,23 +103,25 @@ public void processRecord(
         rejectCommand(command, RejectionType.INVALID_STATE, NO_EVENT_OCCURRED_MESSAGE);
 
       } else {
-        // TODO (saig0): reuse the subscription record in the state (#6533)
-        subscriptionRecord
-            .setElementId(subscription.getTargetElementId())
-            .setInterrupting(subscription.shouldCloseOnCorrelate());
+        // avoid reusing the subscription record directly as any access to the state (e.g. as #get)
+        // will overwrite it - safer to just copy its values into an one-time-use record
+        final ProcessMessageSubscriptionRecord subscriptionRecord = subscription.getRecord();
+        record
+            .setElementId(subscriptionRecord.getElementIdBuffer())
+            .setInterrupting(subscriptionRecord.isInterrupting());
 
```

Comment:
I imagine from the TODO this was not the original plan - however, I think it's safer to do it like this. It's tricky that accessing the state changes the record (for example, in the appliers), very easy to forget. I think in the long term it makes sense to use the record stored in the state as a template in the subscription message handler, but not here, such that the record you get here would already contain all the info you need, and you wouldn't need to fetch the subscription here.

We can do that once subscriptions have keys I guess and we refactor the message handler/API.